nav2-usability/scenario-code/sign-in-routing/router.dart

```diff
@@ -0,0 +1,309 @@
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+class Credentials {
+  final String username;
+  final String password;
+
+  Credentials(this.username, this.password);
+}
+
+abstract class Authentication {
+  bool isSignedIn();
+
+  void signOut();
```

Comment:
Sign out is usually an async operation as well, but it is less important in this use case